js/client.js

```diff
@@ -1325,8 +1325,8 @@
 				if (this.curSideRoom === oldRoom) this.curSideRoom = room;
 				if (this.sideRoom === oldRoom) this.sideRoom = room;
 			}
-			if (type === BattleRoom) this.roomList.push(room);
-			if (type === ChatRoom || type === BattlesRoom) this.sideRoomList.push(room);
+			if (type === BattleRoom && !room.isSideRoom) this.roomList.push(room);
+			if (type === ChatRoom || type === BattlesRoom || (type === BattleRoom && room.isSideRoom)) this.sideRoomList.push(room);
```

Comment:
Actually, what I said wasn't quite right in retrospect, what I mean is that those rooms are not added to the room lists at all. The tabs are added to the tab bar separately, since Home/Ladder/Teambuilder have their own section of the tab bar, and the Rooms room is a special case that should always be to the right of all other side room tabs so separate logic is required for that anyway.